File: extract_pdf.js
const fs = require('fs');
const pdf = require('pdf-parse');

let dataBuffer = fs.readFileSync('d:\\\\IP 2026\\\\Paper I\\\\THE PO REGULATIONS, 2024 FINAL.pdf');

pdf(dataBuffer).then(function(data) {
    fs.writeFileSync('d:\\\\IP 2026\\\\po_regulations_2024_text.txt', data.text);
    console.log('PDF text extracted to d:\\\\IP 2026\\\\po_regulations_2024_text.txt');
});

File: study-planner/scripts/data_fixes/extract_pdf.py
from pypdf import PdfReader
import sys

pdf_path = r"d:\IP 2026\Paper I\Preservation Period of Records.pdf"
output_path = r"d:\IP 2026\preservation_records_full_text.txt"

try:
    reader = PdfReader(pdf_path)
    text = ""
    for page in reader.pages:
        text += page.extract_text() + "\n"
    
    with open(output_path, "w", encoding="utf-8") as f:
        f.write(text)
    print(f"Successfully extracted text to {output_path}")
except Exception as e:
    print(f"Error: {e}")
    sys.exit(1)
